projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -110,7 +110,7 @@ export class IgxGridNavigationService {
 
     public headerNavigation(event: KeyboardEvent) {
         const key = event.key.toLowerCase();
-        if (!HEADER_KEYS.has(key)) {
+        if (!HEADER_KEYS.has(key) || Object.keys(this.activeNode).length === 0) {
```

Comment:
Okay, I initially thought this can't possibly be the check, but it seems like it sort of is as there's already a precedent in the service doing the same thing.

@onlyexeption You should also check if `this.activeNode` is defined first, because some parts of the service reset the object
https://github.com/IgniteUI/igniteui-angular/blob/dc33a249f7180d361d61dcb95486fe4529d494e9/projects/igniteui-angular/src/lib/grids/grid-base.directive.ts#L3486
but some parts also set it to null:
https://github.com/IgniteUI/igniteui-angular/blob/dc33a249f7180d361d61dcb95486fe4529d494e9/projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts#L136

Reply:
Actually a two-in-one check could be `this.activeNode?.row !== -1` - as far as I can tell header nav always sit on row -1.